libsquoosh/src/codecs.ts

```diff
@@ -1,6 +1,37 @@
 import { promises as fsp } from 'fs';
 import { instantiateEmscriptenWasm, pathify } from './emscripten-utils.js';
 
+interface RotateModuleInstance {
+  exports: {
+    memory: WebAssembly.Memory;
+    rotate(width: number, height: number, rotate: number): void;
+  };
+}
+
+interface ResizeWithAspectParams {
+  input_width: number;
+  input_height: number;
+  target_width: number;
+  target_height: number;
+}
+
+interface ResizeInstantiateOptions {
+  width: number;
+  height: number;
+  method: string;
+  premultiply: boolean;
+  linearRGB: boolean;
+}
+
+declare global {
+  // Needed for being able to use ImageData as type in codec types
+  type ImageData = typeof import('./image_data');
```

Comment:
```suggestion
  type ImageData = typeof import('./image_data.js');
```

Reply:
Please check other imports in the rest of the files as well to make sure that they have `.js` :)

Reply:
Ups, sorry for missing them. I now updated all of them :)